scene/resources/bit_map.cpp

```diff
@@ -641,7 +643,8 @@ TypedArray<PackedVector2Array> BitMap::_opaque_to_polygons_bind(const Rect2i &p_
 }
 
 void BitMap::resize(const Size2i &p_new_size) {
-	ERR_FAIL_COND(p_new_size.width < 0 || p_new_size.height < 0);
+	ERR_FAIL_COND_MSG(p_new_size.width < 0, vformat("The requested BitMap width (%d pixels) must be at least 1 pixel.", p_new_size.width));
+	ERR_FAIL_COND_MSG(p_new_size.height < 0, vformat("The requested BitMap width (%d pixels) must be at least 1 pixel.", p_new_size.height));
```

Comment:
These check against 0 yet it says "at least 1 pixel", the error here should be different, as 0 is allowed (though it will error below if it is and the original size isn't 0)

Reply:
I'm trying to follow the logic in `resize()` but I fail to understand it. Why is a width or height of 0 allowed in the first place?

Reply:
I'm not quite sure myself, not sure what happens when you do use it

Reply:
In my blissful ignorance I assume 0 means the BitMap is not fully initialised.

Here you use "pixel" instead of "value". Which should it be, then? I personally think pixels are easier to envision even if inaccurate to the inner workings.